src/services/transaction_reversal.rs

```diff
@@ -14,18 +20,202 @@ pub struct TransactionReversalPayload<'mpesa> {
+        self.receiver_identifier_type = Some(receiver_identifier_type);
+        self
+    }
+
+    // Adds `ResultUrl` This is a required field
+    ///
+    /// # Error
+    /// If `ResultUrl` is invalid or not provided
+    pub fn result_url(mut self, result_url: &'mpesa str) -> Self {
+        self.result_url = Some(result_url);
+        self
+    }
 
-pub struct TransactionReversalResponse;
+    /// Adds `QueueTimeoutUrl` and `ResultUrl`. This is a required field
+    ///
+    /// # Error
+    /// If either `QueueTimeoutUrl` and `ResultUrl` is invalid or not provided
+    pub fn timeout_url(mut self, timeout_url: &'mpesa str) -> Self {
+        self.timeout_url = Some(timeout_url);
+        self
+    }
+
+    /// Comments that are sent along with the transaction.
+    ///
+    /// This is required field
+    pub fn remarks(mut self, remarks: &'mpesa str) -> Self {
+        self.remarks = Some(remarks);
+        self
+    }
+
+    /// Adds any additional information to be associated with the transaction.
+    /// This is an optional Parameter
+    pub fn occasion(mut self, occasion: &'mpesa str) -> Self {
+        self.occasion = Some(occasion);
+        self
+    }
+
+    /// Adds an `amount` to the request
+    ///
+    /// This is a required field
+    pub fn amount<Number: Into<f64>>(mut self, amount: Number) -> Self {
+        self.amount = Some(amount.into());
+        self
+    }
+
+    /// # Transaction Reversal API
+    ///
+    /// Requests for transaction reversal
+    ///
+    /// This API enables reversal of a B2B, B2C or C2B M-Pesa transaction
+    /// Required  parameters:
+    ///
+    /// `transaction_id`: This is the Mpesa Transaction ID of the transaction which you wish to reverse
+    ///
+    /// `amount` : The amount transacted in the transaction to be reversed , down to the cent
+    ///
+    /// `receiver_party`: Your organization's short code.
+    ///
+    /// See more from the Safaricom API docs [here](https://developer.safaricom.co.ke/Documentation)
+    ///
+    /// A successful request returns a `TransactionReversalResponse` type
+    ///
+    /// # Errors
+    /// Returns a `MpesaError` on failure.
+    pub async fn send(self) -> MpesaResult<TransactionReversalResponse> {
+        let url = format!(
+            "{}/mpesa/reversal/v1/request",
+            self.client.environment().base_url()
+        );
+
+        let credentials = self.client.gen_security_credentials()?;
+
+        let payload = TransactionReversalPayload {
+            initiator: self.initiator,
+            security_credentials: &credentials,
+            command_id: self.command_id.unwrap_or(CommandId::TransactionReversal),
+            transaction_id: self
+                .transaction_id
+                .ok_or(MpesaError::Message("transaction_id is required field"))?,
+            receiver_party: self
+                .receiver_party
+                .ok_or(MpesaError::Message("receiver_party is required field"))?,
```

Comment:
Good addition, this is needed for other builders, can be done in another PR

Reply:
Yes, will take up that task.

Reply:
I think we should add a type for this error, so we don't have to type same message everytime.